### PR TITLE
Upload image

### PR DESCRIPTION
**Pull request by assem-gh**

### CODE REVIEW

File: backend/src/main/java/com/elshop/backend/product/ProductController.java [hunk 1]
```diff
@@ -1,40 +1,58 @@
...
-        return new ResponseEntity<>(service.createNewProduct(newProductData), HttpStatus.OK);
+    @PostMapping(consumes = {MediaType.APPLICATION_JSON_VALUE, MediaType.MULTIPART_FORM_DATA_VALUE})
+    public ResponseEntity<Product> createNewProduct(@RequestParam(value = "images") MultipartFile[] imageFiles,
+                                                    @RequestPart("data") @Valid ProductRequest newProductData) {
+
+        String id = appUtils.generateUuid();
+        List<String> imagesUrl = s3Service.uploadMultipleFiles(imageFiles, "products", id);
+        Product product = productService.createNewProduct(newProductData, id, imagesUrl);
+
+        return new ResponseEntity<>(product, HttpStatus.OK);
```
Comment (ChristianBorn) [inline backend/src/main/java/com/elshop/backend/product/ProductController.java]: There is quite a lot of logic in this controller class. Consider putting some of the operations in your service

File: backend/src/main/java/com/elshop/backend/product/ProductService.java [hunk 2]
```diff
@@ -49,4 +49,13 @@ public ProductsListResponse getAll(int page, int size) {
                 pageProduct.getContent()
         );
     }
+
+
+    public void deleteProduct(String id) {
+        Optional<Product> optionalProduct = productRepository.findById(id);
+        if (optionalProduct.isEmpty()) {
+            throw new ResourceNotFoundException("Product", id);
```
Comment (ChristianBorn) [inline backend/src/main/java/com/elshop/backend/product/ProductService.java:57]: Shouldn't this exception be caught somewhere in your controller?

File: frontend/src/store/api/productService.ts [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
 
 const createNewProduct = createAsyncThunk(
   "products/post",
-  async (args: NewProductType, thunkAPI) => {
-    const res = await axiosInstance.post<ProductModel>("/products", args);
+  async (args: any, thunkAPI) => {
+    const res = await axiosInstance.post<ProductModel>("/products", args, {
+      headers: { "Content-Type": "multipart/form-data" },
+    });
+    console.log("[Create Product thunk]: ", res.data);
```
Comment (ChristianBorn) [inline frontend/src/store/api/productService.ts:12]: Remove console output

**Review by ChristianBorn — DISMISSED** [review]

**Comment by ChristianBorn** [comment]
You should increase your coverage, quality gate failed

File: backend/src/main/java/com/elshop/backend/product/ProductService.java (hunk 2 above)
Comment (assem-gh) [reply backend/src/main/java/com/elshop/backend/product/ProductService.java:57]: It caught globally by  AppControllerAdvisor and there I handle exception properly.

**Pull request merged**